MovingBlocks/Terasology
# Rendering overhaul: HazeNode (was SkyBandsNode)

This PR: 
* refactors the SkyBandsNode class into an HazeNode class inheriting from a  generic BlurNode class.
* refactors the NodeFactory class to be more flexible.
* removes a number of redundant OpenGL-related directives.
* Adds some JavaDocs.

This PR does not address newly opened issue #2528.

### How to test
* Create/Load a game and toggle on/off "fog" in the VideoSettings to see the difference with/without haze. 

[Haze ](https://en.wikipedia.org/wiki/Haze) is an atmospheric phenomena blending faraway objects into the color of the atmosphere, or, in Terasology's case, into the color of the sky. Terasology's approach is not completely realistic (it doesn't take in account the shadows cast by faraway mountains for example) but it generally does the trick.
